tests/core/editable/insertion.js

```diff
@@ -159,7 +159,79 @@
 			editor.insertHtml( '<p>bam</p>', 'html', range );
 
 			assert.isInnerHtmlMatching( '<p id="p1">foo</p><p>bam^@</p><p>bar@</p>',
-				bender.tools.selection.getWithHtml( editor ), { compareSelection: true, normalizeSelection: true } );
+				tools.selection.getWithHtml( editor ), { compareSelection: true, normalizeSelection: true } );
 		}
-	} );
+	};
+
+	// Add tests for (#2813).
+	addInsertionTests( [
+		{
+			name: 'test insert when selection starts at the beginning of span',
+			initial: '<span>{foo}bar</span>',
+			data: '<div>div</div>',
+			expected: '<div>div</div><span>bar</span>'
+		}, {
+			name: 'test insert when selection ends at the end of span',
+			initial: '<span>foo{bar}</span>',
+			data: '<div>div</div>',
+			expected: '<span>foo</span><div>div</div>'
+		}, {
+			name: 'test insert when selection covers span',
+			initial: '<span>{foobar}</span>',
+			data: '<div>div</div>',
+			expected: '<div>div</div>'
+		}, {
+			name: 'test insert when selection is in the middle of span',
+			initial: '<span>foo{bar}&nbsp;</span>',
+			data: '<div>div</div>',
+			expected: '<span>foo</span><div>div</div><span>&nbsp;</span>'
+		}
+	] );
+
+	bender.test( tests );
+
+	function addInsertionTests( tests ) {
+		CKEDITOR.tools.array.forEach( tests, addInsertionTest );
```

Comment:
`tests` variable already exists in this file. I know that function is very small, nevertheless, maybe you could rename it to avoid using the same name? Hoisting makes name covering a bit unreadable.